### PR TITLE
feat (log): Custom .log file names

### PR DESCRIPTION
**Pull request by mellobacon**
Addresses #122 and added appropriate docs. 

Probably should have error checking for the file name but just checking if the implementation is good for now.

### CODE REVIEW

**Comment by FabianLars** [comment]
> Probably should have error checking for the file name but just checking if the implementation is good for now.

I think it makes sense to rename the app_name var to log_name to make the intention more clear. Maybe also merge the if into the let declaration so it doesn't even request the app name when there's a custom name, also would get rid of the mutability.

It looks good otherwise. And i'm not sure if we even have to validate it since that's something we generally don't really do with paths/file names 🤔

**Comment by FabianLars** [comment]
Oh and thanks for contributing of course!!

**Comment by mellobacon** [comment]
> I think it makes sense to rename the app_name var to log_name to make the intention more clear. Maybe also merge the if into the let declaration so it doesn't even request the app name when there's a custom name, also would get rid of the mutability.

Will do!
> Oh and thanks for contributing of course!!

And no problem :)

**Comment by FabianLars** [comment]
Ah one last thing i overlooked (besides the format check failing) is to use an [Option](https://doc.rust-lang.org/std/option/index.html) instead of an empty string.
For what it's worth, this would also enable users to name their log file just `.log` if they want to.

**Comment by mellobacon** [comment]
Let me know if there's anything else I missed 😅

File: plugins/log/src/lib.rs [hunk 1]
```diff
@@ -183,12 +184,14 @@ impl Default for Builder {
...
             ))
         });
+        let log_name = None;
         Self {
             dispatch,
             rotation_strategy: DEFAULT_ROTATION_STRATEGY,
             timezone_strategy: DEFAULT_TIMEZONE_STRATEGY,
             max_file_size: DEFAULT_MAX_FILE_SIZE,
             targets: DEFAULT_LOG_TARGETS.into(),
+            log_name,
```
Comment (FabianLars) [inline plugins/log/src/lib.rs]:
```suggestion
        Self {
            dispatch,
            rotation_strategy: DEFAULT_ROTATION_STRATEGY,
            timezone_strategy: DEFAULT_TIMEZONE_STRATEGY,
            max_file_size: DEFAULT_MAX_FILE_SIZE,
            targets: DEFAULT_LOG_TARGETS.into(),
            log_name: None,
```

File: plugins/log/src/lib.rs [hunk 2]
```diff
@@ -262,6 +265,29 @@ impl Builder {
...
+    /// let builder = Builder::default()
+    ///     .targets([
+    ///         LogTarget::LogDir
+    ///     ])
+    ///     .log_name(name)
+    ///     .build()
+    /// ); // Outputs content to custom-name.log
+    ///
+    /// ```
+    pub fn log_name(mut self, log_name: &str) -> Self {
```
Comment (FabianLars) [inline plugins/log/src/lib.rs]:
```suggestion
    pub fn log_name<S: Into<String>>(mut self, log_name: S) -> Self {
```

File: plugins/log/src/lib.rs [hunk 3]
```diff
@@ -284,7 +310,11 @@ impl Builder {
         plugin::Builder::new("log")
             .invoke_handler(tauri::generate_handler![log])
             .setup(move |app_handle| {
-                let app_name = &app_handle.package_info().name;
+                let name = self
+                    .log_name
+                    .clone()
+                    .unwrap_or_else(|| app_handle.package_info().name.clone());
+                let log_name = name.as_str();
```
Comment (FabianLars) [inline plugins/log/src/lib.rs]:
```suggestion
                let log_name = self
                    .log_name
                    .as_deref()
                    .unwrap_or_else(|| &app_handle.package_info().name);
```

**Review by FabianLars — CHANGES_REQUESTED** [review]
Just some small things left because i'm so picky lol (sorry) - meant to push this directly but didn't have access 🤷

File: plugins/log/src/lib.rs [hunk 4]
```diff
@@ -262,6 +264,29 @@ impl Builder {
...
+    ///     .targets([
+    ///         LogTarget::LogDir
+    ///     ])
+    ///     .log_name(name)
+    ///     .build()
+    /// ); // Outputs content to custom-name.log
+    ///
+    /// ```
+    pub fn log_name<S: Into<String>>(mut self, log_name: S) -> Self {
+        self.log_name = Some(log_name.to_string());
```
Comment (FabianLars) [inline plugins/log/src/lib.rs]:
```suggestion
        self.log_name = Some(log_name.into());
```

**Review by FabianLars — APPROVED** [review]
thank youuu

**Pull request merged**